src/main/java/de/rwth/idsg/steve/config/SecurityConfiguration.java

```diff
@@ -100,23 +97,107 @@ public WebSecurityCustomizer webSecurityCustomizer() {
+                    // users
+                    .antMatchers(prefix + "/users").hasAnyRole("USER", "ADMIN")
+                    .antMatchers(prefix + "/users" + "/details/**").hasAnyRole("USER", "ADMIN")
+                     //ocppTags
+                    .antMatchers(prefix + "/ocppTags").hasAnyRole("USER", "ADMIN")
+                    .antMatchers(prefix + "/ocppTags" + "/details/**").hasAnyRole("USER", "ADMIN")
+                     // chargepoints
+                    .antMatchers(prefix + "/chargepoints").hasAnyRole("USER", "ADMIN")
+                    .antMatchers(prefix + "/chargepoints" + "/details/**").hasAnyRole("USER", "ADMIN")
+                     // transactions and reservations
+                    .antMatchers(prefix + "/transactions").hasAnyRole("USER", "ADMIN")
+                    .antMatchers(prefix + "/transactions" + "/details/**").hasAnyRole("USER", "ADMIN")
+                    .antMatchers(prefix + "/reservations").hasAnyRole("USER", "ADMIN")
+                    .antMatchers(prefix + "/reservations" + "/**").hasRole("ADMIN")
+                     // singout and noAccess
+                    .antMatchers(prefix + "/signout/" + "**").hasAnyRole("USER", "ADMIN")
+                    .antMatchers(prefix + "/noAccess/" + "**").hasAnyRole("USER", "ADMIN")
+                     // any other site
+                    .antMatchers(prefix + "/**").hasRole("ADMIN")
+                )
             .sessionManagement(
-                req -> req.invalidSessionUrl(prefix + "/signin")
-            )
+                 req -> req
+                    .invalidSessionUrl(prefix + "/signin")
+                )
             .formLogin(
-                req -> req.loginPage(prefix + "/signin").permitAll()
-            )
+                req -> req
+                    .loginPage(prefix + "/signin")
+                    .permitAll()
+                )
             .logout(
-                req -> req.logoutUrl(prefix + "/signout")
-            )
+                req -> req
+                    .logoutUrl(prefix + "/signout")
+                )
+
+            .exceptionHandling(
+                req -> req
+                    .accessDeniedPage(prefix + "/noAccess")
+                )
             .build();
     }
 
+    /**
+     *
+     * @param auth
+     * @throws Exception
+     */
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth)
+      throws Exception {
+        auth.jdbcAuthentication()
+            .passwordEncoder(CONFIG.getAuth().getPasswordEncoder())
+            .dataSource(dataSource)
+            .usersByUsernameQuery("select username,password,enabled from webusers where username = ?")
+            .authoritiesByUsernameQuery("select username,authority from webauthorities where username = ?");
+    }
+
+
+    /**
+     *
+     * @return
+     */
+    @Bean
+    public UserDetailsManager authenticateUsers() {
+      JdbcUserDetailsManager users = new JdbcUserDetailsManager(dataSource);
```

Comment:
To be discussed but I think a home-made `JooqUserDetailsManager` is not a difficult task and will avoid an extra dependency.

Reply:
agreed. it could then replace `WebUserRepositoryImpl` and some of the other classes.

i think a direct Jooq impl of UserDetailsManager, UserDetailsService that use the classes UserDetails (and User) would nicely align this impl with the concepts of Spring... and therefore even make the home-made classes such as WebUser redundant.

Reply:
[here is my quick attempt](https://github.com/steve-community/steve/commit/df2f9aa1bc21969426b3d6aba554214d0e43ba0d)

Reply:
[addendum](https://github.com/steve-community/steve/commit/3491594b7b4898cda5d41c883cb76a8fa688a77a) to fine-tune details. it is in working condition now.